entry.go

```diff
@@ -208,13 +200,47 @@ func (te *TunaEntry) StartReverse(stream *smux.Stream) error {
 			return
 		}
 		stream.Close()
+
+		totalCost := common.Fixed64(0)
+		lastComputed := common.Fixed64(0)
+		lastClaimed := common.Fixed64(0)
+		lastUpdate := time.Now()
+		claimInterval := time.Duration(te.config.ReverseClaimInterval) * time.Second
+		onErr := nkn.NewOnError(1, nil)
+		isClosed := false
+
+		npc, err := te.Wallet.NewNanoPayClaimer(int32(claimInterval/time.Millisecond), onErr, te.config.ReverseBeneficiaryAddr)
+		if err != nil {
+			log.Println(err)
+			te.close()
+			return
+		}
+
+		go claimCheck(session, npc, onErr, &isClosed)
+
+		go paymentCheck(session, claimInterval, &lastComputed, &lastClaimed, &lastUpdate, &isClosed)
+
 		for {
-			_, err = session.AcceptStream()
+			stream, err = session.AcceptStream()
 			if err != nil {
 				log.Println("Close connection:", err)
 				te.close()
 				return
 			}
+			if len(stream.Metadata()) == 0 {
+				in := atomic.LoadUint64(&te.reverseBytesIn)
+				out := atomic.LoadUint64(&te.reverseBytesOut)
+				if in == 0 && out == 0 {
+					continue
+				}
+				totalCost += te.entryToExitPrice*common.Fixed64(out)/TrafficUnit + te.exitToEntryPrice*common.Fixed64(in)/TrafficUnit
+				err = nanoPayClaim(stream, npc, &lastComputed, &lastClaimed, &totalCost, &lastUpdate)
+				if err != nil {
+					log.Println("nanoPayClaim failed:", err)
+					te.close()
```

Comment:
No need to close on single claim failure. claimCheck and paymentCheck will close te if not updating payment for a while